File: typescript-version/src/family/parent.ts
import { Person, Gender } from './person';
import { Child } from './child';

/**
 * Base class for parents (Father and Mother)
 */
export abstract class Parent extends Person {
  protected numChildren: number = 0;
  protected children: Child[] = [];

  constructor(name: string, gender: Gender) {
    super(name, gender);
  }

  /**
   * Add a child to this parent
   */
  addChild(child: Child): void {
    this.children.push(child);
    this.numChildren += 1;
  }

  /**
   * Get all children
   */
  getChildren(): Child[] {
    return this.children;
  }

  /**
   * Get the number of children
   */
  getNumChildren(): number {
    return this.numChildren;
  }
}

File: python-version/family/parent.py
"""
Parent class - base class for Father and Mother
"""

from typing import List
from .person import Person, Gender


class Parent(Person):
    """Base class for parents (Father and Mother)"""
    
    def __init__(self, name: str, gender: Gender):
        """Initialize a parent"""
        super().__init__(name, gender)
        self.num_children = 0
        self.children: List['Child'] = []
    
    def add_child(self, child: 'Child'):
        """Add a child to this parent"""
        self.children.append(child)
        self.num_children += 1
    
    def get_children(self) -> List['Child']:
        """Get all children"""
        return self.children
    
    def get_num_children(self) -> int:
        """Get the number of children"""
        return self.num_children
